File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/ServerTypes/Tanimlamalar/DoctorsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DoctorsRow } from "./DoctorsRow";
import { Gender } from "./Gender";

export interface DoctorsColumns {
    DoctorId: Column<DoctorsRow>;
    Username: Column<DoctorsRow>;
    Name: Column<DoctorsRow>;
    Surname: Column<DoctorsRow>;
    Gender: Column<DoctorsRow>;
    BirthDate: Column<DoctorsRow>;
    HospitalName: Column<DoctorsRow>;
    DepartmentName: Column<DoctorsRow>;
}

export class DoctorsColumns extends ColumnsBase<DoctorsRow> {
    static readonly columnsKey = 'Tanimlamalar.Doctors';
    static readonly Fields = fieldsProxy<DoctorsColumns>();
}

[Gender]; // referenced types

File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/Tanimlamalar/Doctors/DoctorsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace MuayeneYonetimPortali.Tanimlamalar.Columns;

[ColumnsScript("Tanimlamalar.Doctors")]
[BasedOnRow(typeof(DoctorsRow), CheckNames = true)]
public class DoctorsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int DoctorId { get; set; }
    [EditLink]
    public string Username { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Surname { get; set; }
    public Gender Gender { get; set; }
    public DateTime BirthDate { get; set; }
    public string HospitalName { get; set; }
    public string DepartmentName { get; set; }
}